src/main/java/com/minecolonies/coremod/entity/ai/mobs/barbarians/AbstractEntityBarbarian.java

```diff
@@ -141,14 +141,19 @@ protected ResourceLocation getLootTable()
         return BarbarianSpawnUtils.getBarbarianLootTable(this);
     }
 
+    /**
+     * Should the barbs despawn.
+     * @return true if so.
+     */
+    private boolean shouldDespawn()
+    {
+        return worldTimeAtSpawn != 0 && (worldObj.getTotalWorldTime() - worldTimeAtSpawn) >= 1000;
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Assign this magic number 1000 to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)